src/sqlancer/mysql/gen/MySQLDeleteGenerator.java

```diff
@@ -40,17 +43,28 @@ private SQLQueryAdapter generate() {
         // TODO: support partitions
         sb.append(" FROM ");
         sb.append(randomTable.getName());
+        boolean includeExpressionErrors = false;
         if (Randomly.getBoolean()) {
             sb.append(" WHERE ");
             sb.append(MySQLVisitor.asString(gen.generateExpression()));
             MySQLErrors.addExpressionErrors(errors);
+            includeExpressionErrors = true;
         }
         errors.addAll(Arrays.asList("doesn't have this option",
                 "Truncated incorrect DOUBLE value" /*
                                                     * ignore as a workaround for https://bugs.mysql.com/bug.php?id=95997
                                                     */, "Truncated incorrect INTEGER value",
                 "Truncated incorrect DECIMAL value", "Data truncated for functional index"));
-        // TODO: support ORDER BY
+
+        if (Randomly.getBoolean() && !randomTable.getColumns().isEmpty()) {
+            sb.append(" ORDER BY ");
+            List<MySQLExpression> orderBys = gen.generateOrderBys();
+            sb.append(orderBys.stream().map(exp -> MySQLVisitor.asString(exp)).collect(Collectors.joining(" , ")));
+            if (!includeExpressionErrors) {
```

Comment:
The logic seems wrong here. Like this, we'll never add errors from the `WHERE` clause above.